### DISCUSSION TITLE
Multiple listener ports

### DISCUSSION THREAD

**Discussion by mcbaguetti** [bluenviron/mediamtx#2135] [Questions] [answered]
Hi all, is it possible to have multiple listener ports?

My idea is to stream two or three cameras and share the stream with different HLS ports. 
If I add more cameras in the future, maybe having only one HLS listener port (now it's :8888) would not be so scalable (a single port could be overloaded by the streaming traffic).

One idea I had is to replicate inside docker MediaMtx. change the HLS port inside mediamtx.yml and use that to stream with a different port. The downgrade of this approach would be to have 20 MediaMtx if in the future I will need to output 20 different streams.

**Comment by aler9** [comment, marked as answer]
Hello, you can already use Docker port forwarding to create multiple listeners and forward all incoming connection to the HLS listener:

```
docker run --rm -it \
-p 8888:8888 \
-p 8887:8888 \
-p 8886:8888 \
bluenviron/mediamtx
```

All incoming TCP connections from 8888, 8889, 8886 will be forwarded to 8888.

> having only one HLS listener port (now it's :8888) would not be so scalable (a single port could be overloaded by the streaming traffic).

Unfortunately using multiple ports doesn't improve scalability, you have to launch multiple MediaMTX instances on different PCs / VMs, each responsible of a group of cameras.

**Reply by mcbaguetti** [reply]
thanks! So it wouldn't be useful to have multiple ports for scalability, even if I would have like 50 readers on the same port.

**Reply by aler9** [reply]
Use CloudFlare or Amazon CloudFront in front of the server, or alternatively an home-made storm of HTTP proxies with Squid or Nginx.

**Comment by aler9** [comment]
Alternatively if you want to scale in terms of readers (not cameras), you can use a CDN like CloudFlare or Amazon CloudFront placed in front of MediaMTX.

[a comment by an automated account is not shown]
